src/Oracle/Protocol/OracleHttpProtocol.cs

```diff
@@ -40,7 +40,7 @@ public byte[] Request(ulong requestId, string url, string filter)
             if (!AllowedFormats.Contains(result.Result.Content.Headers.ContentType.MediaType)) throw new InvalidOperationException("ContentType it's not allowed");
 
             var taskRet = result.Result.Content.ReadAsStringAsync();
-            if (!taskRet.Wait(Timeout)) throw new InvalidOperationException("Timeout");
+            if (!taskRet.Wait(Timeout / 2)) throw new InvalidOperationException("Timeout");
```

Comment:
I am not sure if this is fair. It could spend less time than Timeout, and fail.

Reply:
Fixed